src/python_testing/TC_OCC_2_2.py

```diff
@@ -0,0 +1,129 @@
+
+        self.step(1)
+        attributes = Clusters.OccupancySensing.Attributes
+        attribute_list = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.AttributeList)
+
+        self.step(2)
+        if attributes.OccupancySensorType.attribute_id in attribute_list:
+            occupancy_sensor_type_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.OccupancySensorType)
+            feature_map = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.attributes.FeatureMap)
+
+            if (feature_map != feature_map_PIR) & (feature_map != feature_map_US) & (feature_map != feature_map_PHY):
+                asserts.assert_equal(occupancy_sensor_type_dut,
+                                     Clusters.OccupancySensing.Enums.OccupancySensorTypeEnum.kPir, "OccupancySensorType is not PIR")
+            elif (feature_map == feature_map_PIR) & (feature_map != feature_map_US) & (feature_map != feature_map_PHY):
+                asserts.assert_equal(occupancy_sensor_type_dut,
+                                     Clusters.OccupancySensing.Enums.OccupancySensorTypeEnum.kPir, "OccupancySensorType is not PIR")
+            elif (feature_map != feature_map_PIR) & (feature_map == feature_map_US) & (feature_map != feature_map_PHY):
+                asserts.assert_equal(
+                    occupancy_sensor_type_dut, Clusters.OccupancySensing.Enums.OccupancySensorTypeEnum.kUltrasonci, "OccupancySensorType is not Ultrasonic")
+            elif (feature_map == feature_map_PIR) & (feature_map == feature_map_US) & (feature_map != feature_map_PHY):
+                asserts.assert_equal(occupancy_sensor_type_dut, Clusters.OccupancySensing.Enums.OccupancySensorTypeEnum.kPIRAndUltrasonic,
+                                     "OccupancySensorType is not PIRAndUltrasonic")
+            elif (feature_map != feature_map_PIR) & (feature_map != feature_map_US) & (feature_map == feature_map_PHY):
+                asserts.assert_equal(occupancy_sensor_type_dut, Clusters.OccupancySensing.Enums.OccupancySensorTypeEnum.kPhysicalContact,
+                                     "OccupancySensorType is not PhysicalContact")
+            elif (feature_map == feature_map_PIR) & (feature_map != feature_map_US) & (feature_map == feature_map_PHY):
+                asserts.assert_equal(occupancy_sensor_type_dut,
+                                     Clusters.OccupancySensing.Enums.OccupancySensorTypeEnum.kPir, "OccupancySensorType is not PIR")
+            elif (feature_map != feature_map_PIR) & (feature_map == feature_map_US) & (feature_map == feature_map_PHY):
+                asserts.assert_equal(
+                    occupancy_sensor_type_dut, Clusters.OccupancySensing.Enums.OccupancySensorTypeEnum.kUltrasonic, "OccupancySensorType is not Ultrasonic")
+            elif (feature_map == feature_map_PIR) & (feature_map == feature_map_US) & (feature_map == feature_map_PHY):

```

Comment:
these equalities cannot happen together: this is a bitmap so if PIR and ULTRASONIC is true, then neither == for PIR nor PHY nor US will happen ... because the bits are set, not the equality.

This whole logic has to be re-worked.

Reply:
Change the logic statement as requested and added comments. Done.